pom.xml

```diff
@@ -93,7 +93,7 @@
                 <version>${quartz.version}</version>
                 <exclusions>
                     <exclusion>
-                        <groupId>c3p0</groupId>
+                        <groupId>com.mchange</groupId>
```

Comment:
Can we remove c3p0 when quartz upgrade?

Reply:
quartz 2.3.0 has two default implementations of pooling connection provider: c3p0 and HikaryCP, so if we don't use c3p0 in ElasticJob, we can safely remove the dependency from ElasticJob pom. So quartz does have a compile dependency to c3p0, but not mendatory for runtime.

Reply:
The license of c3p0 is LGPL which is not friendly for Apache license, please use HikariCP instead of it

Reply:
I'm a bit confused. Did you mean we are using quartz persistence in ElasticJob, which can persist trigger info to db? But actually, ElasticJob doesn't use quartz persistence capability, therefore c3p0 never exist in our pom at all (by adding <exclusion> under quartz dependency node in pom.xml, as seen in above code). This means we don't need use HikariCP currently, unless we add trigger persistence feature later.

Reply:
Quartz 2.2.1 only provides a c3p0 implementation of pooling connection provider, without capability of customized extension. While in 2.3.0, a 'PoolingConnectionProvider' interface and two implementations (c3p0 and HicariCP) are provided, so we have choices in case we need store trigger info.

Reply:
No, we do not need use c3p0, and do not need persist job into database.
So we should remove c3p0's dependency.
Please use HikariCP to instead of c3p0 for license reason